en_US/file-transfer/client.md

```diff
@@ -0,0 +1,88 @@
+
+
+## File Transfer Commands
+
+All file transfer commands are regular MQTT PUBLISH messages with QOS 1 sent to specific topics.
+
+### Init Command
+
+The `init` command is used to initialize a file transfer session.
+
+Topic: `$file/{file_id}/init`
+
+Payload: a JSON object with the following fields:
+```
+{
+  "name": "{name}",
+  "size": {size},
+  "checksum": "{checksum}"
+  "expire_at": {expire_at}
+  "segments_ttl": {segments_ttl}
+  "user_data": {user_data}
+}
+```
+
+* `file_id` is the unique identifier of the file transfer session. The client device generates it, which must be unique across all file transfer sessions.
+* `name` is the name of the file. It will be sanitized using percent-encoding if it coincides with reserved file names "." and ".."; if it contains "/", "\", "%", ":" or unprintable unicode characters. The binary length of the name must not exceed 240 bytes.
+* `size` is the size of the file in bytes. This is an informational field and is not used by the broker.
+* `checksum` is the SHA256 checksum of the file. It is optional, but if provided, the broker will verify the file's checksum after it is uploaded.
+* `expire_at` is the timestamp in seconds since the epoch when the file may be deleted from the storage.
+* `segments_ttl` is the time-to-live of the file segments in seconds. The value will be clamped to the range `minimum_segments_ttl` to `maximum_segments_ttl` configured in the broker.
+* `user_data` is an arbitrary JSON object that will be stored along with the file metadata. It can be used to store any additional information about the file.
+
+In the payload, the only required field is `name`.
+
+Example:
+```json
+{
+  "name": "ml-logs-data.log",
+  "size": 12345,
+  "checksum": "1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef",
+  "expire_at": 1696659943,
+  "segments_ttl": 600
+}
+```
+
+### Segment Command
+
+The `segment` command is used to upload a chunk of the file.
+
+Topic: `$file/{file_id}/{offset}[/{checksum}]`
+
+Payload: the binary data of the file chunk.
+
+* `file_id` is the unique identifier of the file transfer session.
+* `offset` is the offset in bytes from the beginning of the file where the chunk should be written.
+
+### Finish Command
+
+The `finish` command is used to finish the file transfer session.
+
+Topic: `$file/{file_id}/fin/{file_size}[/{checksum}]`
+
+Payload: not used.
+
+* `file_id` is the unique identifier of the file transfer session.
+* `file_size` is the size of the whole file in bytes.
+* `checksum` is the SHA256 checksum of the whole file. If specified, this value has a greater priority than the `checksum` field in the `init` command.
+
+The broker will verify that it has all the segments to assemble the file and start exporting to the configured exporter. If the file is successfully exported and its checksum is valid, the broker will respond with success RC. Otherwise, the broker will respond with an error.
```

Comment:
I hope to provide as many examples or working SDKs as possible to help users learn the funciton.
```suggestion
The broker will verify that it has all the segments to assemble the file and start exporting to the configured exporter. If the file is successfully exported and its checksum is valid, the broker will respond with success RC. Otherwise, the broker will respond with an error.

## SDK & Example

- [Python](https://github.com/emqx/emqx-ft/blob/main/src/emqx_ft/main.py)
```